pkg/server/datastore/sqlstore/sqlstore.go

```diff
@@ -443,6 +456,33 @@ func (ds *Plugin) PruneRegistrationEntries(ctx context.Context, expiresBefore ti
 	})
 }
 
+// ListRegistrationEntriesEvents lists all registration entry events
+func (ds *Plugin) ListRegistrationEntriesEvents(ctx context.Context, req *datastore.ListRegistrationEntriesEventsRequest) (resp *datastore.ListRegistrationEntriesEventsResponse, err error) {
+	if err = ds.withReadTx(ctx, func(tx *gorm.DB) (err error) {
+		resp, err = listRegistrationEntriesEvents(tx, req)
+		return err
+	}); err != nil {
+		return nil, err
+	}
+	return resp, nil
+}
+
+// PruneRegistrationEntriesEvents deletes all registration entry events older than a specified duration (i.e. more than 24 hours old)
+func (ds *Plugin) PruneRegistrationEntriesEvents(ctx context.Context, olderThan time.Duration) (err error) {
+	return ds.withWriteTx(ctx, func(tx *gorm.DB) (err error) {
```

Comment:
Since the createdAt timestamp is supplied by the server (gorm populates it via time.Now), there doesn't seem to be a need to try and do the equality check using the database notion of "now()" and the variant-specific type coercions and functions? I haven't tried it, but I wonder if the following works for all variants:

```
 tx.Where("created_at < ?", now.Add(-olderThan)).Delete(&RegisteredEntryEvent{})
```

Reply:
Thanks, that worked.